plugins/examples/rest-handler/build.gradle

```diff
@@ -29,6 +29,22 @@ esplugin {
   noticeFile rootProject.file('NOTICE.txt')
 }
 
+configurations {
+  restSpec
+}
+
+dependencies {
+  restSpec project(':rest-api-spec')
```

Comment:
Understood, I think the refactoring you suggested will remove the need for this. 

> It's not even clear why this is necessary. Folks building plugins should not need to run our rest api tests.

There is a project with an example REST test, which needs the needs the REST spec. This was an attempt to avoid making a reference to the rootBuildDir. Moot point though :)